### DISCUSSION TITLE
future of petie-vue

### DISCUSSION THREAD

**Discussion by sqllyw** [vuejs/petite-vue#53] [General]
Hi,

did not see any update in a month, what's the direction of this framework?

**Comment by niutech** [comment]
I see a big potential in Petite Vue. It could be a modern and very lightweight replacement for jQuery with syntax almost the same as Vue.js, encouraging progressive enhancement, ensuring good SEO, without a hassle with SSR or prerendering. Only in the week of 20-26.07 there were almost 10k downloads from npm, so it has some gain. @yyx990803 could you please continue to support it?

**Comment by pthethanh** [comment]
I have the same question. I used alpinejs before but finally switched to petitle-vue and loves it so much, but I'm a little bit worry about its future since I don't see any new changes since the last month.

**Reply by sqllyw** [reply]
In the exact situation, now my project has two "sprinkle" js libraries, alpine and petie-vue, I'm  using alpine for any new pages added into the project until petite-view future is clear, I prefer 'mpa' over 'spa' and sprinkle type js is enough for mpa app.

**Comment by TomS-** [comment]
A question I can answer!

He released this "early", it was a proof of concept for Evan. It's completely his intention to support it - just not yet. He is focusing on some features for Vue first.

**Reply by sqllyw** [reply]
the proof of concept works so well, but if Evan has to focus on Vue, better leave this to Alpine

**Comment by wbijker** [comment]
I can't really understand the need for Alpine or Vue Petite if standard Vue already has all the features already built in. Size cannot be a factor as these are all below 100Kb. 

With little effort (120 loc) I built my own version of Vue Petite. Used it for quite some while now.  I've packaged the full Vue 2 experience  into a small utility that can be useful for non-bundler projects, or for progressive enhancement in Evan's terms.

[Link to Vue-Hydrate](https://github.com/itechodev/vue-hydrate)

Most simple example:

```html
<!DOCTYPE html>
<html>
<head>
    <script src="https://unpkg.com/vue-hydrate@1.0/dist/vue-hydrate-bundle.js"></script>
</head>
<body>
    
    <div v-data="{message: 'It works.'}">
        <span>{{ message }} </span>
    </div>
    
</body>
</html>
```

**Reply by wbijker** [reply]
[The classic todo list example build in vue-hydrate](https://github.com/itechodev/vue-hydrate/blob/master/web/examples/todo.html)

**Reply by niutech** [reply]
Cool, but it looks like a clone of Petite Vue. What's the advantage of Vue Hydrate?

As for Vue vs Petite Vue, the former is full-blown and does not provide progressive enhancement without prerendering or SSR. The latter fits elegantly if you want to sprinkle some interactions on a static HTML web page.

**Reply by sqllyw** [reply]
I'm back to coding after six months absent!

As a 'occasional' developer, I have programs developed many many years ago, and never update it until several years later, with those old languages and tools(delphi/java), I can easily back to the development mode and be productive again, with the new SPA tools like Angular/Vue, if I do not use it everyday and get back to the code half year later, I am a total newbie, I have to re-learn so many concepts before I can start coding, this usually take quite a while, longer than learning vue/angular as you have a  whole project to review with your 'newly learned' knowledge and admire yourself for creative way of implementing this and that features that seems totally new now to yourself.

with petie-vue, it took a very short time to 're-learn', maybe one day, and I'm back to development mode, start to update the program. that's the reason I like petie-vew or alphinejs,  usually projects with petie-vue are MPA, backend is expressjs or java, which are easier to manage.

that's my experience with petie-vue/alphine, of course for  full stack developers you code everyday, you don't have issues like this, so a vuejs offers more features for you to utilize.

**Reply by wbijker** [reply]
@niutech, I think having the same API may be advantageous. Say one day you want to move your code to SPA then little has to be done - it all works the same.  And more obvious Vue hydrate supports everything Vue does like transitions, components etc which Vue petite does not.

I have used Vue hydrate in private capacity long ago - before Vue petite - and  later published it to NPM to address the issue of just sprinkle interactivity on statically served pages. And it worked great as an alternative to JQuery and others. 

I'm somewhat surprised that we didn't see more usages of Vue template compiler being shipped to the browser to enable this. I guess alpine is great,  definitely not original, but  Vue already contained everything alpine has.

**Reply by m2017atTR** [reply]
please can we stop reinventing the wheel again and again ... nearly ten years ago angularjs without spa. after type script wheel invented with typescript finished projects.And AngularJsMaterial was there (open source)
Favor of Angular, angularjs ended support is the problem...

[https://material.angularjs.org/latest/demo/autocomplete](https://material.angularjs.org/latest/demo/autocomplete)
[https://docs.angularjs.org/guide/bootstrap](https://docs.angularjs.org/guide/bootstrap)

**Comment by tipiirai** [comment]
The question is indeed valid. Is this project abandoned?

**Comment by sqllyw** [comment]
i don't think it's abandoned, the amazing part of this project is, it's so small, it just works, does not require too much learning and improvements.

**Reply by tipiirai** [reply]
I agree that the project is great. It's just lacking developers. Nobody is maintaining it.

**Reply by sqllyw** [reply]
the doc says : Driven by @vue/reactivity, and the latter also being used by Alpinejs and vue.

**Reply by tipiirai** [reply]
Well.. if you just look at this repository: the dates of commit and the presence of developers on this discussion area — the project seems abandoned. Obviously vue/reactivity is not abandoned, but that's just dependency.

**Reply by sqllyw** [reply]
right, almost six months and commits are not merged, but it also means the project just works without too many frequent updates.

**Reply by tipiirai** [reply]
Seems so. However, people are reluctant to switch to a project whose future is unclear. Just like you said above 

> I'm using alpine for any new pages added into the project until petite-view future is clear

Also: there are quite many bugs and open issues on this repository.

**Reply by sqllyw** [reply]
after v0.4.0, I do not encounter bugs, have  been using it since then

**Comment by pwFoo** [comment]
I compared petite-vue and alpinejs and decided to play with petite-vue because of size and syntax...
But would be good to know if it is still maintained.... 🤔

**Reply by solution-loisir** [reply]
I guess that it is maintained within the scope of what's described in https://github.com/vuejs/petite-vue#status. So that would mean:
* Until v1.0.0, use at your own risk.
* No features requests accepted for now.
* A team will probably lead the project eventually.

I'm still investing in petite-vue vs Alpine because of the orientation of the project being progressive enhancement vs a replacement for jquery and a companion to SSR rendering framework like Livewire. To be fair, Alpine can do progressive enhancement just as well, but the bundle size is getting bloated these days. Plus, I like that the syntax will be kept as close as possible to Vue.

**Comment by sqllyw** [comment]
The [doc](https://vuejs.org/guide/extras/ways-of-using-vue.html#standalone-script) in vuejs.org mentions petite-vue as part of vue ecosystem, I do not see why petite-vue is not being maintained, it might not have frequent updates, but it does not have any major issues anyway:

Vue also provides an alternative distribution called [petite-vue](https://github.com/vuejs/petite-vue) that is specifically optimized for progressively enhancing existing HTML. It has a smaller feature set, but is extremely lightweight and uses an implementation that is more efficient in no-build-step scenarios.

**Comment by ThePlayerSD** [comment]
The docs state this is still a part of the ecosystem so hopefully it's going to be maintained 🙏

**Reply by halivert** [reply]
Yeah, also the [neovim volar extension have support for petite-vue](https://github.com/yaegassy/coc-volar#vitepress-and-petite-vue-support) 🎉

**Reply by sqllyw** [reply]
even just merging a simple pull request is enough to let us know it is still being maintained.

**Comment by jaredcwhite** [comment]
A couple of thoughts: some of the relevant code powering petite-vue is in @vue/reactivity so we know for sure that will continue to get attention. Also, I've skimmed through a lot of the code of petite-vue proper, and it's remarkably concise and straightforward (kudos to @yyx990803 et al.!). So even on the slim chance that year(s) from now nobody's updating this repo, I'm not too concerned about a community-maintained fork picking up the slack.

I'm currently working on a library to embed petite-vue into custom elements for a nice marriage of web components with reactive HTML, so if need be I'm fine with building petite-vue and shipping it directly with the library. But obviously I'd prefer to see petite-vue continue to receive official updates and version bumps, even if just for a bug fix here and there.

**Reply by terryaney** [reply]
Speaking of building.  I've cloned and can rebuild the typescript...but how to I build the project so that it creates the same 'single IIFE file' found at https://unpkg.com/petite-vue@0.4.1/dist/petite-vue.iife.js?

**Reply by terryaney** [reply]
Figured out `npm run build`

**Comment by sqllyw** [comment]
pvue works without any update almost a year! 
Are we expecting a update before 2022 ends? 
at least we will not have a so out of date (11 months) pvue for 2023:)

**Comment by sqllyw** [comment]
to @yyx990803,

just merge a commit will do, this version number seems good, always forgot what version I'm in:

 feat: add version display #128 

let's start 2023 with a up to date Pettie-vue:)

**Comment by sqllyw** [comment]
Interesting numbers:

Petite-vue:
15 months, 107 commits, average 7 commits per month, latest commit 11 months ago, 7.2k stars

AlpineJS:
24 months, 1858 commits, average 77 commits per month, latest commit 4 days ago, 22.4k stars

Petie-vue has almost no update in the year, but still works. However  latest commit 11months vs 4 days really not encouraging for any new projects coming.

**Reply by dy** [reply]
why do you expect a project to constantly have commits? Good things don't need changes

**Comment by Hiws** [comment]
There recently was a Q&A on the VueLand Discord, where Evan answered a question regarding `petite-vue`.

Figured it was relevant to this discussion.
<https://discord.com/channels/325477692906536972/1050439284341096571/1054597576101474384>

> petite-vue is indeed intended to fill the gap for progressive enhancement cases where Vue 3 would be too heavy-handed.
> 
> It is not abandoned, but rather it is considered "done" because the scope is well defined. I don't think it needs more features (as that would defeat the purpose of being lean and minimal). If you find yourself needing more than what petite-vue provides, you can either go up to Vue proper, or try https://alpinejs.dev/.
> 
> That said, I should update the README to indicate this more clearly.

**Comment by xinjie-zhang** [comment]
I have the same concern if petite-vue has been abandoned. I planned to build a headless ui library on petite-vue, just like I've done for alpinejs, [Vimesh Headless](https://github.com/vimeshjs/vimesh-headless). Current petite-vue need more improvements for serious web application. Anyway, do not just stop it.

**Comment by tipiirai** [comment]
I wrote a super minimal (2.3k gzipped) Vue/Petite-Vue alternative that is designed from ground up for server-side rendering and progressive enhancement. It's called [Nue JS](https://nuejs.org/docs/nuejs/) and it's a central part of a bigger [ecosystem](https://nuejs.org/ecosystem/) to ultimately replace many of the current/heavyweight/complex tools that cause frontend fatigue. Nue is under active development.

**Reply by sqllyw** [reply]
tried and nice, but still need some more time to understand.

**Reply by volkandkaya** [reply]
I was pretty interested, but doesn't seem to be similar and not sure what the goal is of Nue.

Also read

"Yes, but it's only the Nue runtime which is 2.3KB, the compiler bundle is 36KB. Petite Vue includes everything in 7KB."

Petite Vue is just magically when working with it.

I think the docs of Nue need redoing to make it crystal clear what is going on. Maybe read the readme of Petite Vue for inspiration.

**Reply by jikkuatwork** [reply]
@tipiirai Is there a way Nue can be used without bundling? I don't mind the size, but I hate the `node_modules` workflow.

**Reply by niutech** [reply]
@jikkuatwork It has been already asked: https://github.com/nuejs/nue/issues/85 and the answer is **no**.

**Reply by jikkuatwork** [reply]
Cool! Thanks!

**Comment by alexbruno** [comment]
Is `petite-vue` still relevant?

Today I'm facing a problem in enterprise web apps that should be nice to have a light framework like this one, but it looks kind of dead, with the latest update more than one year and a half ago.

I'm already a "full Vue.js" lover, but comparing `petite-vue` with other alternatives I'm not sure about it.

For exemple, AlpineJS and Solid are up and running with active development and receiving recent updates.

**Reply by niutech** [reply]
There are updated forks of Petite Vue: [Pico Vue](https://github.com/wusaby-rush/pico-vue) (with plugins) and [Petite Vue Pro](https://github.com/604587986/petite-vue-pro) (with $watch).

**Reply by nfplee** [reply]
I've been using it absolutely fine for sometime now. Apart from missing the watch command, I haven't had much trouble porting most of my components across from the full framework.

It also has a few benefits for me, inline templates (removed in Vue 3) is great for server-side rendering and the fact you can have multiple instances of v-scope (including nesting them) is essential for my application.

The fact it hasn't been updated for sometime doesn't bother me as it's driven by @vue/reactivity and I've not encountered any bugs.

**Reply by volkandkaya** [reply]
It is amazing how solid the core is. I'm looking at extending but not sure everyone would need the "plugins" so no point forking it.

**Comment by DominusKelvin** [comment]
I think petite-vue really relevant for the same use case you'd use JQuery or Alpine.

The fact there is no "updates" seem to me because at this time it's feature complete and you can use it as is

**Comment by Phil-DS** [comment]
I've been a fan of this little library. Saw it mentioned a few years ago, and its nice how out of the box and familiar it feels. Would be a shame to see it die.

**Reply by DominusKelvin** [reply]
> I've been a fan of this little library. Saw it mentioned a few years ago, and its nice how out of the box and familiar it feels. Would be a shame to see it die.

I don't think it's going to die

**Reply by sqllyw** [reply]
it works very well after two years without any update, maybe it's time to release it as version 1.0?

**Reply by niutech** [reply]
@sqllyw First @yyx990803 please merge [bug fixing PRs](https://github.com/vuejs/petite-vue/pulls) before releasing version 1.0.

**Reply by jikkuatwork** [reply]
Yes, please consider the PRs & please enable `Issues`.

**Comment by alexbruno** [comment]
I just figured out that maybe petite-vue is not necessary anymore.
Building a huge website with HTML pages compiled with server-side logic, I've applied standard Vue over the existent HTML, using the "enhancing power" as Vue is a progressive framework.
I have styles built with PostCSS + Tailwind and scripts with TypeScript, everything through a bundler, so I have full-control over the custom stack and I can build the scripts with all the possible optimizations, like tree-shaking.
So, standard Vue is still better than petite-vue in this case and more powerful.
I think petite-vue is relevant only if you don't have a bundler setup and will use JS as is, with no build step.

**Reply by niutech** [reply]
For huge websites Petite Vue may not be useful, but for small ones it is. Petite Vue doesn't require a build step and it is only 16 KB minified, whereas Vue is 97 KB minified.

**Reply by DominusKelvin** [reply]
> I just figured out that maybe petite-vue is not necessary anymore.
> 
> Building a huge website with HTML pages compiled with server-side logic, I've applied standard Vue over the existent HTML, using the "enhancing power" as Vue is a progressive framework.
> 
> I have styles built with PostCSS + Tailwind and scripts with TypeScript, everything through a bundler, so I have full-control over the custom stack and I can build the scripts with all the possible optimizations, like tree-shaking.
> 
> So, standard Vue is still better than petite-vue in this case and more powerful.
> 
> I think petite-vue is relevant only if you don't have a bundler setup and will use JS it as is, with no build step.

Yeah use Petite-Vue for the same use case as Alpine

**Reply by alexbruno** [reply]
Agree, but standard Vue is 97kb only if you use the whole thing with no optimizations, like a bundler with tree-shaking extracting only the functions you really use in your code at the build step.

**Reply by DominusKelvin** [reply]
> Agree, but standard Vue is 97kb only if you use the whole thing with no optimizations, like a bundler with tree-shaking extracting only the functions you really use in your code at the build step.

Yeah I get that but from a more server centric approach which is what petite-vue was made for, I'll just pick it

I'm working on a project here at Sails.js which will have petite-Vue as a first party for adding interactivity to server side templates

**Reply by sqllyw** [reply]
for me, size is not the issue, petite-vue force me to write simpler codes, and that's enough,  I have the full version Vue standby, in case I have some pages that require more interactions I can just simply use vuejs , however I have yet to find a use case that petite-vue can not handle. of course all depends on the type of apps you are developing, for me it's just a simple MPA app, server renders html using a handlebar template, and petite-vue take over in the browser

**Reply by sqllyw** [reply]
more to that, i use the buildless approach, so no bundler.

**Reply by ishaanparlikar** [reply]
> Agree, but standard Vue is 97kb only if you use the whole thing with no optimizations, like a bundler with tree-shaking extracting only the functions you really use in your code at the build step.

Can you explain how do you achieve this at the build step? 
Probably share your vite config?

**Reply by lucafabbian** [reply]
Yes, it is. If you work for a modern tech company and youa re building a traditional website, you'll never gonna use it. But that's not always the case. There are a lot of scenarios where you cannot have a build process, either because people are not able to deal with it, or because you are generating the website on the fly. I really hope that @yyx990803 will (at least) merge the bug fixes pull requests.

**Reply by alexbruno** [reply]
Unfortunately guys, I must say that even for no-build cases, there are better options, like Solid.js for example.
In fact, I'm dropping Vue even for the build cases and adopting Solid.js.

**Reply by volkandkaya** [reply]
How are you using Solidjs without a build step? Doesn't it use JSX?

**Reply by niutech** [reply]
There is [solid-web.js](https://github.com/yangzhaox/solid-web), but it is 49KB minified, compared with just 16KB minified petite-vue.

**Reply by volkandkaya** [reply]
view-source:https://yangzhaox.github.io/solid-web/ don't think this is solving the same problem Petite Vue is.

**Reply by niutech** [reply]
True, Solid.js doesn't provide progressive enhancement and it is not a good alternative for Petite Vue, I was just pointing out that it is theoretically possible to run Solid.js without a build step.
For updated forks of Petite Vue, check out: [PPVue](https://github.com/PSanilP/ppvue), [Pico Vue](https://github.com/wusaby-rush/pico-vue) (with plugins) and [Petite Vue Pro](https://github.com/604587986/petite-vue-pro) (with $watch).

**Reply by alexbruno** [reply]
It is not theoretically. Did you guys read the example in the main official page?
[#buildless-options](https://www.solidjs.com/guides/getting-started#buildless-options)

**Comment by sqllyw** [comment]
last update was 2022-01-18, 0.41, looks like not any update in 2023?

**Reply by DominusKelvin** [reply]
> last update was 2022-01-18, 0.41, looks like not any update in 2023?

I don't there is need for an update as the scope for petite-vue is pretty much covered as it is.

**Reply by sqllyw** [reply]
That reminded me of Linux servers in the early years of 2000, a server can be up in one or two years without rebooting, Petie-vue is in that category:)

**Reply by alexbruno** [reply]
Without rebooting and with no security updates or bug fixes, running with security issues for 2 years.

**Reply by sqllyw** [reply]
that's the good old days, once configured and up, no need to touch

**Comment by lucafabbian** [comment]
To have bug fixes implemented? Moreover, the api is missing some key
features.

**Comment by ilahazs** [comment]
So when author comeback to this project?

**Comment by sqllyw** [comment]
I'm doubtful about author coming back

**Reply by jikkuatwork** [reply]
I just wish he considered the PRs, opened up Issues & started a specific channel for PV in their discord. This is the least he can do!

**Comment by dy** [comment]
In the meantime I'd invite to check out [sprae](https://github.com/dy/sprae).

**Reply by meirlamdan** [reply]
I recommend checking out [alpineJs](https://alpinejs.dev/) as well

**Reply by jikkuatwork** [reply]
Have tried AlpineJS. But there is a killer feature for PV: Components. I have been working on some interesting components and workflow styles in the past month and is gearing up to a release, will show what PV can actually do!

**Reply by niutech** [reply]
@meirlamdan AlpineJS is 43KB minified and Petite Vue is 16KB minified, so they're not in the same category. Full Vue is 40KB minified, so it is on par with AlpineJS.

**Reply by dy** [reply]
@niutech pssst sprae is [9kb minified](https://github.com/dy/sprae/blob/main/dist/sprae.min.js), and fastest in class

**Comment by sqllyw** [comment]
finally petite-vue reached 9k stars despite almost three years no update!

**Comment by jikkuatwork** [comment]
Wish they atleast opened up the issues.On 29 Jul 2024, at 4:49 AM, sqllyw ***@***.***> wrote:﻿
finally petite-vue reached 9k stars despite almost three years no update!

**Reply by sqllyw** [reply]
agreed, actually no need to do anything, just merge those pull requests .

**Comment by jikkuatwork** [comment]
Yes! I am wondering, whether a fork would be possible. I can’t believe this is only being by a hand full of people.On 30 Jul 2024, at 4:33 AM, sqllyw ***@***.***> wrote:﻿
agreed, actually no need to do anything, just merge those pull requests .

**Comment by verybigelephants** [comment]
there are some bugs that would need addressing i wish the makers came back? i mean, felt so liberated when i swithced to petite vue from full vue and react. the speed, no bundling, rapid prototyping and fully sufficient for simpler frontend

**Reply by jikkuatwork** [reply]
Absolutely!!! I am not going back at all. I am even willing to work on this in production with bugs compared to the previous React bundling workflow. I simply can't.

**Reply by niutech** [reply]
In the meantime, you can use a more up-to-date fork of Petite Vue, like [Pico Vue](https://github.com/ws-rush/pico-vue) or [We-Make-Websites' fork](https://github.com/we-make-websites/petite-vue/tree/development).

**Reply by jikkuatwork** [reply]
Thanks! Pico Vue looks nice. I wish it was just merged into the main one.

**Reply by verybigelephants** [reply]
> In the meantime, you can use a more up-to-date fork of Petite Vue, like [Pico Vue](https://github.com/ws-rush/pico-vue) or [We-Make-Websites' fork](https://github.com/we-make-websites/petite-vue/tree/development).

thanks for the tip, however, from the commits it looks like the the forks haven't gotten much momentum, they have 1/2 semi-active developers each and there's not much of a progress done

i wish some fork has gained traction and/or publicity, as these forks seem like rather obscure inactive forks

**Comment by alexbruno** [comment]
Do you know what?
I'm just wondering...
What if this project was abandoned because it has potential to surpass Vue itself in popularity?

**Reply by jikkuatwork** [reply]
I was telling my team exactly this jokingly...! But wait!

**Reply by sqllyw** [reply]
very likely, browsers nowadays have more features, what needed is this 'reactivity' which petite-view fits well, then again, the Signal Proposal (in stage 1 now) in the TC39 (https://github.com/tc39/proposal-signals) might take this place too once approved, that's the day framework less?

**Reply by jikkuatwork** [reply]
Great! I just wish PV just matured to 1.0 and just remained there. A very set of features and a few issues needs to be solved for it to be the go to tool.

**Comment by meirlamdan** [comment]
It's really unfortunate that there is no update from the author @yyx990803 on the status of this project. If it has been decided not to continue with it, at least let us know.

**Reply by Hiws** [reply]
He has mentioned that he considers the project "done"/"feature complete":
https://github.com/vuejs/petite-vue/discussions/53#discussioncomment-4483647

He also talked about it in a recent Podcast:
https://www.youtube.com/watch?v=OG8dvyKLjIM&t=1094s

**Reply by alexbruno** [reply]
But what about the issues?

**Reply by verybigelephants** [reply]
Yeah, he could at least accept pull requests and address the bugs

**Comment by sqllyw** [comment]
Preac/Signal + VanilaJS = Petite-vue?

**Reply by dy** [reply]
(psst preact-signal + vanillajs = [sprae](https://github.com/dy/sprae))
Vue has its own signals.

**Reply by sqllyw** [reply]
yes, but thinking of a petie-vue replacement, petie-vue is lightweight, so does Preac/Signal

**Reply by dy** [reply]
give it a try. It is faster, smaller and simpler. And it's easy to switch back to petite-vue.

**Reply by ishaanparlikar** [reply]
Why not svelte or solid at this point? If you like vue syntax then svelte. Both are very lightweight.

**Reply by sqllyw** [reply]
> Why not svelte or solid at this point? If you like vue syntax then svelte. Both are very lightweight.

I'd consider svelte another language compiled into javascript, it's syntax is more like javascript but it is not a real javascript.

**Comment by sqllyw** [comment]
[using preact/signal for something like pvue count example](https://jsfiddle.net/kz7pcmgt/)

**Reply by verybigelephants** [reply]
that one is going to have a terrible performance on a real project https://www.youtube.com/watch?v=INLq9RPAYUw

**Comment by jikkuatwork** [comment]
The most fascinating thing about Petite Vue for me is the ability to use it without a bundler. Just import a single file are you are done!I know jQuery & few others can work like this; but its ability to do proper components that can also be imported like this makes it killer!—JJOn 11 Feb 2025, at 3:31 PM, sqllyw ***@***.***> wrote:﻿

Why not svelte or solid at this point? If you like vue syntax then svelte. Both are very lightweight.

I'd consider svelte another language compiled into javascript, it's syntax is more like javascript but it is not a real javascript.

**Reply by okainov** [reply]
You can also use full Vue.js without any bundler, just import a single file and done :)

**Comment by sqllyw** [comment]
actually it's just searching a petite-vue alternative for a future project.

I do like to stay in petite-vue in the future projects, I know petite-vue is 'feature complete' and it works for so many years without updating, but I do like to use one that is actively maintained which petite-vue is not:

petite-vue is built on top of two major components in the vue ecosystem, namely:

1) reactivity
https://www.npmjs.com/package/@vue/reactivity
current version is 3.5.13, pvue version is 3.2.27

2) vite
https://www.npmjs.com/package/vite
current version is 6.1.0, pvue version is 2.7.12

no need new features, just update it to work with those new versions

**Reply by verybigelephants** [reply]
it's not "feature complete", it's full of bugs. the owner just said that because admitting that you're done with a project because you're not interested anymore is too hard

**Reply by sqllyw** [reply]
many alternatives to pvue nowadays, my app is more a MPA type, any new pages added, I use other approaches while keeping pvue in the existing pages, works great. for new projects I would not use pvue.

**Reply by dy** [reply]
@sqllyw which ones? Alpinejs, lucia (deprecated) and sprae are the only ones I know of.

**Reply by sqllyw** [reply]
I think there are two kinds of pvue users:

    Vue users who are already comfortable with the Vue ecosystem.
    Vanilla JS users who just need a bit of reactivity but mostly rely on what the browser can do.

I’m in the second group—I try to keep things simple and let the browser do most of the heavy lifting. For people like me, apps don’t really depend on the whole Vue ecosystem. We just need to sprinkle in some JavaScript using pvue when necessary.

Here are some things I’m using (or planning to use):

    Preact/Signal
    Lit.dev (about to use)
    HTMX (using now)
    Fixi (only 89 lines, but covers most of HTMX’s features)
    HTMZ (super tiny—just 181 bytes—but surprisingly powerful for MPA apps) - using now

That said, if you're using pvue because you're already into Vue, these alternatives might not be for you. As for me, I’ll keep using pvue for my existing pages—it’s actually a really solid framework.

**Comment by alex-jitbit** [comment]
I agree with @wbijker 

If you just want a simple declarative way to add a little bit of reactivity without leaving HTML, you can just... use Vue

```html
<div id="myApp">
	Hello {{ name }}
</div>
<script type="module">
	Vue.createApp({
		data: () => ({ name: "John" })
	}).mount("#myApp")
</script>
```

**Reply by verybigelephants** [reply]
Yyyeah not going to sabotage my website with whooping 1.5MB junk just to "add a little bit of reactivity"

**Reply by niutech** [reply]
@alex-jitbit No, full-blown Vue is much bigger than Petite Vue and doesn't support progressive enhancement.

**Reply by alex-jitbit** [reply]
@verybigelephants `vue.global.prod.min.js` is 130kb and that's _not even gzipped_. Take a closer look at your 1.5mb junk, it's not coming from Vue

**Reply by alex-jitbit** [reply]
Anyways, my reply was to offer another alternative b/c petite-vue is clearly abandoned
